logprep/processor/hyperscan_resolver/processor.py

```diff
@@ -0,0 +1,225 @@
+"""This module contains functionality for resolving log event values using regex lists."""
+
+import errno
+from logging import Logger, DEBUG
+from multiprocessing import current_process
+from os import path, makedirs
+from typing import List
+
+# pylint: disable=no-name-in-module
+from hyperscan import (
+    Database,
+    HS_FLAG_SINGLEMATCH,
+    HS_FLAG_CASELESS,
+    loadb,
+    dumpb,
+)
+
+# pylint: enable=no-name-in-module
+
+from logprep.processor.base.processor import RuleBasedProcessor
+from logprep.processor.hyperscan_resolver.rule import HyperscanResolverRule
+from logprep.util.processor_stats import ProcessorStats
+
+
+class HyperscanResolverError(BaseException):
+    """Base class for HyperscanResolver related exceptions."""
+
+    def __init__(self, name: str, message: str):
+        super().__init__(f"HyperscanResolver ({name}): {message}")
+
+
+class DuplicationError(HyperscanResolverError):
+    """Raise if field already exists."""
+
+    def __init__(self, name: str, skipped_fields: List[str]):
+        message = (
+            "The following fields already existed and "
+            "were not overwritten by the Generic Resolver: "
+        )
+        message += " ".join(skipped_fields)
+
+        super().__init__(name, message)
+
+
+class HyperscanResolver(RuleBasedProcessor):
+    """Resolve values in documents by referencing a mapping list."""
+
+    def __init__(
+        self,
+        name: str,
+        configuration: dict,
+        logger: Logger,
+    ):
+        specific_rules_dirs = configuration.get("specific_rules")
+        generic_rules_dirs = configuration.get("generic_rules")
+        tree_config = configuration.get("tree_config")
+        super().__init__(name, tree_config, logger)
+        self._hyperscan_databases = {}
+
+        hyperscan_db_path = configuration.get("hyperscan_db_path")
+        if hyperscan_db_path:
+            self._hyperscan_database_path = hyperscan_db_path
+        else:
+            self._hyperscan_database_path = f"{path.dirname(path.abspath(__file__))}/hyperscan_dbs/"
+        self.ps = ProcessorStats()
+
+        self._event = None
+
+        self._replacements_from_file = {}
+
+        self.add_rules_from_directory(specific_rules_dirs, generic_rules_dirs)
```

Comment:
```suggestion
        self.add_rules_from_directory(
            specific_rules_dirs=specific_rules_dirs, 
            generic_rules_dirs=generic_rules_dirs)
```
to be more explicit and to ensure the two kinds of rules are not switched

Reply:
not seen?

Reply:
I changed it.